src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Cms.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_CmsDecode")]
+        internal static extern SafeCmsHandle CmsDecode([In] byte[] buf, int len);
```

Comment:
With byte arrays the marshaler shouldn't be making a copy; I believe it just pins the array and passes a pointer to the first element.  There's shouldn't be a copy happening in either direction.